<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="RauENVS">
<title>Phyloseq analysis template • RauENVS</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Phyloseq analysis template">
<meta property="og:description" content="RauENVS">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">RauENVS</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/parse_antismash.html">Parse AntiSmash and BigScape</a>
    <a class="dropdown-item" href="../articles/phyloseq_analysis.html">Phyloseq analysis template</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Phyloseq analysis template</h1>
                        <h4 data-toc-skip class="author">AU-ENVS
Bioinformatics</h4>
            
      
      
      <div class="d-none name"><code>phyloseq_analysis.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://au-envs-bioinformatics.github.io/RauENVS/">RauENVS</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217" class="external-link">phyloseq</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://microbiome.github.io/microbiome" class="external-link">microbiome</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/" class="external-link">cowplot</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vegandevs/vegan" class="external-link">vegan</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/set_theme_AU_ENVS.html">set_theme_AU_ENVS</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"phyloseq_template_toy_data"</span><span class="op">)</span></span></code></pre></div>
<p>This is a RMarkdown template for doing a quick analysis of an OTU and
taxonomic table using RMarkdown.</p>
<p>It contains toy data to illustrate what this notebook does.
<strong>First, make sure you run the entire notebook with the toy
example</strong>. To do this, click on “Knit” in Rstudio.</p>
<div class="section level3">
<h3 id="read-your-data">Read your data<a class="anchor" aria-label="anchor" href="#read-your-data"></a>
</h3>
<p>You need 3 tabulated files with contains metadata, OTU and taxonomy,
respectively. Please make sure you read the data correctly. Delete the
code that generates the toy example and uncomment the code that reads
files.</p>
<div class="section level5">
<h5 id="read-metadata-table">Read metadata table<a class="anchor" aria-label="anchor" href="#read-metadata-table"></a>
</h5>
<p>You metadata table <strong>must</strong> contain a column named
rowname and as many features as desired. Either edit the tabulated file
or rename the column with the sample identifier. Your metadata table
should look something like the example below.</p>
<p>By assigning a variable inside () we print a summary.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># (metadata_tbl &lt;- readr::read_csv("metadata_dummy.csv"))</span></span>
<span><span class="va">metadata_tbl</span> <span class="op">&lt;-</span> <span class="va">phyloseq_template_toy_data</span><span class="op">$</span><span class="va">meta</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">metadata_tbl</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">rowname</th>
<th align="left">layer</th>
<th align="left">age</th>
<th align="right">pH</th>
<th align="right">depth</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">TK20-2</td>
<td align="left">AL</td>
<td align="left">AY</td>
<td align="right">4.02</td>
<td align="right">20</td>
</tr>
<tr class="even">
<td align="left">TK20-3</td>
<td align="left">AL</td>
<td align="left">AY</td>
<td align="right">4.29</td>
<td align="right">30</td>
</tr>
<tr class="odd">
<td align="left">TK20-4</td>
<td align="left">AL</td>
<td align="left">AY</td>
<td align="right">4.25</td>
<td align="right">40</td>
</tr>
<tr class="even">
<td align="left">TK20-5</td>
<td align="left">TZ1</td>
<td align="left">AM</td>
<td align="right">4.64</td>
<td align="right">50</td>
</tr>
<tr class="odd">
<td align="left">TK20-6</td>
<td align="left">TZ1</td>
<td align="left">AM</td>
<td align="right">4.13</td>
<td align="right">60</td>
</tr>
<tr class="even">
<td align="left">TK20-7</td>
<td align="left">TZ1</td>
<td align="left">AO</td>
<td align="right">4.86</td>
<td align="right">70</td>
</tr>
<tr class="odd">
<td align="left">TK20-8</td>
<td align="left">TZ2</td>
<td align="left">AO</td>
<td align="right">4.48</td>
<td align="right">80</td>
</tr>
<tr class="even">
<td align="left">TK20-9</td>
<td align="left">TZ2</td>
<td align="left">AO</td>
<td align="right">4.57</td>
<td align="right">90</td>
</tr>
<tr class="odd">
<td align="left">TK20-10</td>
<td align="left">PF</td>
<td align="left">AO</td>
<td align="right">4.91</td>
<td align="right">100</td>
</tr>
<tr class="even">
<td align="left">TK20-11</td>
<td align="left">AL</td>
<td align="left">AY</td>
<td align="right">4.22</td>
<td align="right">10</td>
</tr>
<tr class="odd">
<td align="left">TK20-12</td>
<td align="left">AL</td>
<td align="left">AY</td>
<td align="right">4.02</td>
<td align="right">20</td>
</tr>
<tr class="even">
<td align="left">TK20-13</td>
<td align="left">AL</td>
<td align="left">AY</td>
<td align="right">4.29</td>
<td align="right">30</td>
</tr>
<tr class="odd">
<td align="left">TK20-14</td>
<td align="left">AL</td>
<td align="left">AY</td>
<td align="right">4.25</td>
<td align="right">40</td>
</tr>
<tr class="even">
<td align="left">TK20-15</td>
<td align="left">TZ1</td>
<td align="left">AM</td>
<td align="right">4.64</td>
<td align="right">50</td>
</tr>
<tr class="odd">
<td align="left">TK20-16</td>
<td align="left">TZ1</td>
<td align="left">AM</td>
<td align="right">4.13</td>
<td align="right">60</td>
</tr>
<tr class="even">
<td align="left">TK20-17</td>
<td align="left">TZ1</td>
<td align="left">AO</td>
<td align="right">4.86</td>
<td align="right">70</td>
</tr>
<tr class="odd">
<td align="left">TK20-18</td>
<td align="left">TZ2</td>
<td align="left">AO</td>
<td align="right">4.48</td>
<td align="right">80</td>
</tr>
<tr class="even">
<td align="left">TK20-19</td>
<td align="left">TZ2</td>
<td align="left">AO</td>
<td align="right">4.57</td>
<td align="right">90</td>
</tr>
<tr class="odd">
<td align="left">TK20-20</td>
<td align="left">PF</td>
<td align="left">AO</td>
<td align="right">4.91</td>
<td align="right">100</td>
</tr>
<tr class="even">
<td align="left">TK20-21</td>
<td align="left">AL</td>
<td align="left">AY</td>
<td align="right">4.22</td>
<td align="right">10</td>
</tr>
<tr class="odd">
<td align="left">TK20-22</td>
<td align="left">AL</td>
<td align="left">AY</td>
<td align="right">4.02</td>
<td align="right">20</td>
</tr>
<tr class="even">
<td align="left">TK20-23</td>
<td align="left">AL</td>
<td align="left">AY</td>
<td align="right">4.29</td>
<td align="right">30</td>
</tr>
<tr class="odd">
<td align="left">TK20-24</td>
<td align="left">AL</td>
<td align="left">AY</td>
<td align="right">4.25</td>
<td align="right">40</td>
</tr>
<tr class="even">
<td align="left">TK20-25</td>
<td align="left">TZ1</td>
<td align="left">AM</td>
<td align="right">4.64</td>
<td align="right">50</td>
</tr>
<tr class="odd">
<td align="left">TK20-26</td>
<td align="left">TZ1</td>
<td align="left">AM</td>
<td align="right">4.13</td>
<td align="right">60</td>
</tr>
<tr class="even">
<td align="left">TK20-27</td>
<td align="left">TZ1</td>
<td align="left">AO</td>
<td align="right">4.86</td>
<td align="right">70</td>
</tr>
<tr class="odd">
<td align="left">TK20-28</td>
<td align="left">TZ2</td>
<td align="left">AO</td>
<td align="right">4.48</td>
<td align="right">80</td>
</tr>
<tr class="even">
<td align="left">TK20-29</td>
<td align="left">TZ2</td>
<td align="left">AO</td>
<td align="right">4.57</td>
<td align="right">90</td>
</tr>
<tr class="odd">
<td align="left">TK20-30</td>
<td align="left">PF</td>
<td align="left">AO</td>
<td align="right">4.91</td>
<td align="right">100</td>
</tr>
</tbody>
</table>
</div>
<div class="section level5">
<h5 id="read-otu-table">Read OTU table<a class="anchor" aria-label="anchor" href="#read-otu-table"></a>
</h5>
<p>Please, make sure you take into account if the table is tabulated
using “,”, “;” or tabs. Your otu table should look something like the
example below.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># otu_tbl &lt;- readr::read_csv("otu_dummy.csv")</span></span>
<span><span class="va">otu_tbl</span> <span class="op">&lt;-</span> <span class="va">phyloseq_template_toy_data</span><span class="op">$</span><span class="va">otu</span></span>
<span><span class="va">otu_tbl</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<colgroup>
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
</colgroup>
<thead><tr class="header">
<th align="right">TK20-2</th>
<th align="right">TK20-3</th>
<th align="right">TK20-4</th>
<th align="right">TK20-5</th>
<th align="right">TK20-6</th>
<th align="right">TK20-7</th>
<th align="right">TK20-8</th>
<th align="right">TK20-9</th>
<th align="right">TK20-10</th>
<th align="right">TK20-11</th>
<th align="right">TK20-12</th>
<th align="right">TK20-13</th>
<th align="right">TK20-14</th>
<th align="right">TK20-15</th>
<th align="right">TK20-16</th>
<th align="right">TK20-17</th>
<th align="right">TK20-18</th>
<th align="right">TK20-19</th>
<th align="right">TK20-20</th>
<th align="right">TK20-21</th>
<th align="right">TK20-22</th>
<th align="right">TK20-23</th>
<th align="right">TK20-24</th>
<th align="right">TK20-25</th>
<th align="right">TK20-26</th>
<th align="right">TK20-27</th>
<th align="right">TK20-28</th>
<th align="right">TK20-29</th>
<th align="right">TK20-30</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">344</td>
<td align="right">338</td>
<td align="right">872</td>
<td align="right">944</td>
<td align="right">187</td>
<td align="right">83</td>
<td align="right">17</td>
<td align="right">47</td>
<td align="right">6</td>
<td align="right">412</td>
<td align="right">774</td>
<td align="right">355</td>
<td align="right">4769</td>
<td align="right">992</td>
<td align="right">2959</td>
<td align="right">18</td>
<td align="right">1</td>
<td align="right">30</td>
<td align="right">0</td>
<td align="right">139</td>
<td align="right">646</td>
<td align="right">363</td>
<td align="right">962</td>
<td align="right">956</td>
<td align="right">884</td>
<td align="right">246</td>
<td align="right">0</td>
<td align="right">23</td>
<td align="right">70</td>
</tr>
<tr class="even">
<td align="right">5776</td>
<td align="right">14725</td>
<td align="right">4922</td>
<td align="right">3000</td>
<td align="right">3304</td>
<td align="right">646</td>
<td align="right">298</td>
<td align="right">448</td>
<td align="right">401</td>
<td align="right">10733</td>
<td align="right">5590</td>
<td align="right">9648</td>
<td align="right">80899</td>
<td align="right">8926</td>
<td align="right">783</td>
<td align="right">234</td>
<td align="right">65</td>
<td align="right">190</td>
<td align="right">71</td>
<td align="right">13144</td>
<td align="right">6740</td>
<td align="right">9200</td>
<td align="right">4332</td>
<td align="right">4624</td>
<td align="right">911</td>
<td align="right">186</td>
<td align="right">4</td>
<td align="right">149</td>
<td align="right">412</td>
</tr>
<tr class="odd">
<td align="right">4</td>
<td align="right">5</td>
<td align="right">3</td>
<td align="right">2</td>
<td align="right">9</td>
<td align="right">295</td>
<td align="right">4380</td>
<td align="right">10308</td>
<td align="right">7</td>
<td align="right">16</td>
<td align="right">13</td>
<td align="right">4</td>
<td align="right">29</td>
<td align="right">0</td>
<td align="right">23</td>
<td align="right">167</td>
<td align="right">1190</td>
<td align="right">5087</td>
<td align="right">3</td>
<td align="right">3</td>
<td align="right">15</td>
<td align="right">11</td>
<td align="right">10</td>
<td align="right">9</td>
<td align="right">1</td>
<td align="right">5</td>
<td align="right">5</td>
<td align="right">8557</td>
<td align="right">16</td>
</tr>
<tr class="even">
<td align="right">83</td>
<td align="right">73</td>
<td align="right">56</td>
<td align="right">11</td>
<td align="right">18</td>
<td align="right">64</td>
<td align="right">125</td>
<td align="right">306</td>
<td align="right">1</td>
<td align="right">84</td>
<td align="right">181</td>
<td align="right">61</td>
<td align="right">387</td>
<td align="right">4</td>
<td align="right">83</td>
<td align="right">42</td>
<td align="right">23</td>
<td align="right">156</td>
<td align="right">0</td>
<td align="right">36</td>
<td align="right">117</td>
<td align="right">52</td>
<td align="right">113</td>
<td align="right">25</td>
<td align="right">16</td>
<td align="right">63</td>
<td align="right">2</td>
<td align="right">43</td>
<td align="right">9</td>
</tr>
<tr class="odd">
<td align="right">1798</td>
<td align="right">886</td>
<td align="right">2251</td>
<td align="right">3176</td>
<td align="right">437</td>
<td align="right">1598</td>
<td align="right">2589</td>
<td align="right">7291</td>
<td align="right">4790</td>
<td align="right">261</td>
<td align="right">3532</td>
<td align="right">785</td>
<td align="right">14943</td>
<td align="right">1829</td>
<td align="right">8676</td>
<td align="right">1151</td>
<td align="right">2294</td>
<td align="right">3138</td>
<td align="right">8</td>
<td align="right">174</td>
<td align="right">2186</td>
<td align="right">555</td>
<td align="right">2508</td>
<td align="right">2096</td>
<td align="right">3313</td>
<td align="right">2074</td>
<td align="right">65</td>
<td align="right">2503</td>
<td align="right">2538</td>
</tr>
<tr class="even">
<td align="right">292</td>
<td align="right">595</td>
<td align="right">540</td>
<td align="right">29</td>
<td align="right">89</td>
<td align="right">75</td>
<td align="right">12</td>
<td align="right">63</td>
<td align="right">11</td>
<td align="right">234</td>
<td align="right">388</td>
<td align="right">362</td>
<td align="right">2452</td>
<td align="right">18</td>
<td align="right">254</td>
<td align="right">28</td>
<td align="right">143</td>
<td align="right">31</td>
<td align="right">0</td>
<td align="right">231</td>
<td align="right">318</td>
<td align="right">414</td>
<td align="right">708</td>
<td align="right">74</td>
<td align="right">25</td>
<td align="right">38</td>
<td align="right">14</td>
<td align="right">67</td>
<td align="right">27</td>
</tr>
</tbody>
</table>
</div>
<div class="section level5">
<h5 id="read-otu-table-1">Read OTU table<a class="anchor" aria-label="anchor" href="#read-otu-table-1"></a>
</h5>
<p>Please, make sure you take into account if the table is tabulated
using “,”, “;” or tabs. Your taxonomic table should look something like
the example below.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># taxonomic_tbl &lt;- readr::read_csv("taxonomic_dummy.csv")</span></span>
<span><span class="va">taxonomic_tbl</span> <span class="op">&lt;-</span> <span class="va">phyloseq_template_toy_data</span><span class="op">$</span><span class="va">taxa</span></span>
<span><span class="va">taxonomic_tbl</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="7%">
<col width="12%">
<col width="24%">
<col width="12%">
<col width="12%">
<col width="10%">
<col width="20%">
</colgroup>
<thead><tr class="header">
<th align="left">Kingdom</th>
<th align="left">Phylum</th>
<th align="left">Class</th>
<th align="left">Order</th>
<th align="left">Family</th>
<th align="left">Genus</th>
<th align="left">Species</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">k__Bacteria</td>
<td align="left">p__Proteobacteria</td>
<td align="left">c__Deltaproteobacteria</td>
<td align="left">o__Oligoflexales</td>
<td align="left">f__0319-6G20</td>
<td align="left">g__</td>
<td align="left">s__</td>
</tr>
<tr class="even">
<td align="left">k__Bacteria</td>
<td align="left">p__Actinobacteria</td>
<td align="left">c__Actinobacteria (class)</td>
<td align="left">o__Kineosporiales</td>
<td align="left">f__Kineosporiaceae</td>
<td align="left">g__Kineosporia</td>
<td align="left">s__Unknown Kineosporia species 1</td>
</tr>
<tr class="odd">
<td align="left">k__Bacteria</td>
<td align="left">p__Actinobacteria</td>
<td align="left">c__Acidimicrobiia</td>
<td align="left">o__Acidimicrobiales</td>
<td align="left">f__OM1 clade</td>
<td align="left">g__</td>
<td align="left">s__</td>
</tr>
<tr class="even">
<td align="left">k__Bacteria</td>
<td align="left">p__Ca. Parcubacteria</td>
<td align="left">c__</td>
<td align="left">o__</td>
<td align="left">f__</td>
<td align="left">g__</td>
<td align="left">s__</td>
</tr>
<tr class="odd">
<td align="left">k__Bacteria</td>
<td align="left">p__Proteobacteria</td>
<td align="left">c__Betaproteobacteria</td>
<td align="left">o__Nitrosomonadales</td>
<td align="left">f__Nitrosomonadaceae</td>
<td align="left">g__</td>
<td align="left">s__</td>
</tr>
<tr class="even">
<td align="left">k__Bacteria</td>
<td align="left">p__Acidobacteria</td>
<td align="left">c__Solibacteres (Acidobacteria group 3)</td>
<td align="left">o__Solibacterales</td>
<td align="left">f__Solibacteraceae</td>
<td align="left">g__Ca. Solibacter</td>
<td align="left">s__</td>
</tr>
</tbody>
</table>
</div>
<div class="section level5">
<h5 id="sanity-check">Sanity check<a class="anchor" aria-label="anchor" href="#sanity-check"></a>
</h5>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">otu_tbl</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">metadata_tbl</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">otu_tbl</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">taxonomic_tbl</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="preprocess-your-data">Preprocess your data<a class="anchor" aria-label="anchor" href="#preprocess-your-data"></a>
</h3>
<div class="section level5">
<h5 id="transform-into-phyloseq-object">Transform into phyloseq object<a class="anchor" aria-label="anchor" href="#transform-into-phyloseq-object"></a>
</h5>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">otu</span> <span class="op">&lt;-</span> <span class="va">otu_tbl</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html" class="external-link">otu_table</a></span><span class="op">(</span>taxa_are_rows <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">taxa</span> <span class="op">&lt;-</span> <span class="va">taxonomic_tbl</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html" class="external-link">tax_table</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="va">metadata_tbl</span><span class="op">|&gt;</span></span>
<span>  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"rowname"</span><span class="op">)</span><span class="op">|&gt;</span></span>
<span>  <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html" class="external-link">sample_data</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Now, we combine OTU table, taxonomy table and metadata into a
<code>phyloseq</code> object.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">physeq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/phyloseq.html" class="external-link">phyloseq</a></span><span class="op">(</span><span class="va">otu</span>, <span class="va">taxa</span>, <span class="va">meta</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; phyloseq-class experiment-level object</span></span>
<span><span class="co">#&gt; otu_table()   OTU Table:         [ 3728 taxa and 29 samples ]</span></span>
<span><span class="co">#&gt; sample_data() Sample Data:       [ 29 samples by 4 sample variables ]</span></span>
<span><span class="co">#&gt; tax_table()   Taxonomy Table:    [ 3728 taxa by 7 taxonomic ranks ]</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="dealing-with-taxonomic-categories">Dealing with taxonomic categories<a class="anchor" aria-label="anchor" href="#dealing-with-taxonomic-categories"></a>
</h5>
<p>We will remove the SILVA prefix, anything labelled “unknown” (this
will not affect abundances) and replace spaces with underscores.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#|label: prepro-taxa</span></span>
<span><span class="co"># Remove prefix</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html" class="external-link">tax_table</a></span><span class="op">(</span><span class="va">physeq</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"Unknown.*"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html" class="external-link">tax_table</a></span><span class="op">(</span><span class="va">physeq</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html" class="external-link">tax_table</a></span><span class="op">(</span><span class="va">physeq</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html" class="external-link">tax_table</a></span><span class="op">(</span><span class="va">physeq</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/remove_silva_prefix_from_taxtable.html">remove_silva_prefix_from_taxtable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="data-cleaning">Data cleaning<a class="anchor" aria-label="anchor" href="#data-cleaning"></a>
</h5>
<p>You may want to do some custom data cleaning here using the function
<code>subset_taxa</code>. As an example, we can remove all taxa which
belong to the SAR Kingdom groups. Notice how the number of taxa have
been reduced.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">physeq</span> <span class="op">&lt;-</span> <span class="va">physeq</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_taxa-methods.html" class="external-link">subset_taxa</a></span><span class="op">(</span><span class="va">Kingdom</span> <span class="op">!=</span> <span class="st">"SAR"</span><span class="op">)</span></span>
<span><span class="va">physeq</span></span>
<span><span class="co">#&gt; phyloseq-class experiment-level object</span></span>
<span><span class="co">#&gt; otu_table()   OTU Table:         [ 3615 taxa and 29 samples ]</span></span>
<span><span class="co">#&gt; sample_data() Sample Data:       [ 29 samples by 4 sample variables ]</span></span>
<span><span class="co">#&gt; tax_table()   Taxonomy Table:    [ 3615 taxa by 7 taxonomic ranks ]</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="making-unique-otu-labels">Making unique OTU labels<a class="anchor" aria-label="anchor" href="#making-unique-otu-labels"></a>
</h3>
<p>The function <code>make_physeq_tax_label_unique</code> would only
work if “empty” taxonomic ranks are encoded as ““. Please, make sure of
it before running the function. Otherwise, you will get ugly unique OTU
labels (in purpose).</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">" "</span>, <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html" class="external-link">tax_table</a></span><span class="op">(</span><span class="va">physeq</span><span class="op">)</span>, fixed<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html" class="external-link">tax_table</a></span><span class="op">(</span><span class="va">physeq</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>First, we create the new labels:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">new_otu_labels</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html" class="external-link">tax_table</a></span><span class="op">(</span><span class="va">physeq</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/make_physeq_tax_label_unique.html">make_physeq_tax_label_unique</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">new_otu_labels</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "0319-6G20"         "Kineosporia"       "OM1clade"         </span></span>
<span><span class="co">#&gt; [4] "Ca.Parcubacteria"  "Nitrosomonadaceae" "Ca.Solibacter"</span></span></code></pre></div>
<p>And, if everything looks fine, we assign the new labels.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_names-methods.html" class="external-link">taxa_names</a></span><span class="op">(</span><span class="va">physeq</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">new_otu_labels</span></span></code></pre></div>
<div class="section level5">
<h5 id="transform-sample-counts">Transform sample counts<a class="anchor" aria-label="anchor" href="#transform-sample-counts"></a>
</h5>
<p>How you transform your counts, or do so, will depend on the
experiment and factors such as the number of replicates.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">physeq_rel</span> <span class="op">&lt;-</span> <span class="va">physeq</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transformcounts.html" class="external-link">transform_sample_counts</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span>
<span>  </span>
<span><span class="va">physeq_mean_rel</span> <span class="op">&lt;-</span>  <span class="va">physeq_rel</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/merge_samples-methods.html" class="external-link">merge_samples</a></span><span class="op">(</span><span class="st">"depth"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transformcounts.html" class="external-link">transform_sample_counts</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>       </span>
<span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_sums.html" class="external-link">sample_sums</a></span><span class="op">(</span><span class="va">physeq_mean_rel</span><span class="op">)</span></span>
<span><span class="co">#&gt;  10  20  30  40  50  60  70  80  90 100 </span></span>
<span><span class="co">#&gt; 100 100 100 100 100 100 100 100 100 100</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="inspect-data">Inspect data<a class="anchor" aria-label="anchor" href="#inspect-data"></a>
</h3>
<p>Now, we can inspect the resulting tables:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">physeq_rel</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html" class="external-link">otu_table</a></span><span class="op">(</span><span class="op">)</span><span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="7%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">TK20-2</th>
<th align="right">TK20-3</th>
<th align="right">TK20-4</th>
<th align="right">TK20-5</th>
<th align="right">TK20-6</th>
<th align="right">TK20-7</th>
<th align="right">TK20-8</th>
<th align="right">TK20-9</th>
<th align="right">TK20-10</th>
<th align="right">TK20-11</th>
<th align="right">TK20-12</th>
<th align="right">TK20-13</th>
<th align="right">TK20-14</th>
<th align="right">TK20-15</th>
<th align="right">TK20-16</th>
<th align="right">TK20-17</th>
<th align="right">TK20-18</th>
<th align="right">TK20-19</th>
<th align="right">TK20-20</th>
<th align="right">TK20-21</th>
<th align="right">TK20-22</th>
<th align="right">TK20-23</th>
<th align="right">TK20-24</th>
<th align="right">TK20-25</th>
<th align="right">TK20-26</th>
<th align="right">TK20-27</th>
<th align="right">TK20-28</th>
<th align="right">TK20-29</th>
<th align="right">TK20-30</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">0319-6G20</td>
<td align="right">0.006</td>
<td align="right">0.007</td>
<td align="right">0.023</td>
<td align="right">0.029</td>
<td align="right">0.007</td>
<td align="right">0.001</td>
<td align="right">0.001</td>
<td align="right">0.001</td>
<td align="right">0.000</td>
<td align="right">0.013</td>
<td align="right">0.007</td>
<td align="right">0.010</td>
<td align="right">0.022</td>
<td align="right">0.044</td>
<td align="right">0.013</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.001</td>
<td align="right">0.000</td>
<td align="right">0.006</td>
<td align="right">0.008</td>
<td align="right">0.012</td>
<td align="right">0.024</td>
<td align="right">0.022</td>
<td align="right">0.022</td>
<td align="right">0.005</td>
<td align="right">0.000</td>
<td align="right">0.001</td>
<td align="right">0.003</td>
</tr>
<tr class="even">
<td align="left">Kineosporia</td>
<td align="right">0.101</td>
<td align="right">0.320</td>
<td align="right">0.129</td>
<td align="right">0.094</td>
<td align="right">0.116</td>
<td align="right">0.007</td>
<td align="right">0.012</td>
<td align="right">0.008</td>
<td align="right">0.015</td>
<td align="right">0.336</td>
<td align="right">0.053</td>
<td align="right">0.272</td>
<td align="right">0.370</td>
<td align="right">0.397</td>
<td align="right">0.003</td>
<td align="right">0.004</td>
<td align="right">0.002</td>
<td align="right">0.007</td>
<td align="right">0.299</td>
<td align="right">0.610</td>
<td align="right">0.088</td>
<td align="right">0.298</td>
<td align="right">0.106</td>
<td align="right">0.105</td>
<td align="right">0.023</td>
<td align="right">0.004</td>
<td align="right">0.004</td>
<td align="right">0.004</td>
<td align="right">0.019</td>
</tr>
<tr class="odd">
<td align="left">OM1clade</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.003</td>
<td align="right">0.181</td>
<td align="right">0.192</td>
<td align="right">0.000</td>
<td align="right">0.001</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.003</td>
<td align="right">0.038</td>
<td align="right">0.189</td>
<td align="right">0.013</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.005</td>
<td align="right">0.235</td>
<td align="right">0.001</td>
</tr>
<tr class="even">
<td align="left">Ca.Parcubacteria</td>
<td align="right">0.001</td>
<td align="right">0.002</td>
<td align="right">0.001</td>
<td align="right">0.000</td>
<td align="right">0.001</td>
<td align="right">0.001</td>
<td align="right">0.005</td>
<td align="right">0.006</td>
<td align="right">0.000</td>
<td align="right">0.003</td>
<td align="right">0.002</td>
<td align="right">0.002</td>
<td align="right">0.002</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.001</td>
<td align="right">0.001</td>
<td align="right">0.006</td>
<td align="right">0.000</td>
<td align="right">0.002</td>
<td align="right">0.002</td>
<td align="right">0.002</td>
<td align="right">0.003</td>
<td align="right">0.001</td>
<td align="right">0.000</td>
<td align="right">0.001</td>
<td align="right">0.002</td>
<td align="right">0.001</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">Nitrosomonadaceae</td>
<td align="right">0.031</td>
<td align="right">0.019</td>
<td align="right">0.059</td>
<td align="right">0.099</td>
<td align="right">0.015</td>
<td align="right">0.018</td>
<td align="right">0.107</td>
<td align="right">0.136</td>
<td align="right">0.175</td>
<td align="right">0.008</td>
<td align="right">0.033</td>
<td align="right">0.022</td>
<td align="right">0.068</td>
<td align="right">0.081</td>
<td align="right">0.038</td>
<td align="right">0.019</td>
<td align="right">0.074</td>
<td align="right">0.117</td>
<td align="right">0.034</td>
<td align="right">0.008</td>
<td align="right">0.029</td>
<td align="right">0.018</td>
<td align="right">0.062</td>
<td align="right">0.048</td>
<td align="right">0.084</td>
<td align="right">0.041</td>
<td align="right">0.065</td>
<td align="right">0.069</td>
<td align="right">0.117</td>
</tr>
<tr class="even">
<td align="left">Ca.Solibacter</td>
<td align="right">0.005</td>
<td align="right">0.013</td>
<td align="right">0.014</td>
<td align="right">0.001</td>
<td align="right">0.003</td>
<td align="right">0.001</td>
<td align="right">0.000</td>
<td align="right">0.001</td>
<td align="right">0.000</td>
<td align="right">0.007</td>
<td align="right">0.004</td>
<td align="right">0.010</td>
<td align="right">0.011</td>
<td align="right">0.001</td>
<td align="right">0.001</td>
<td align="right">0.000</td>
<td align="right">0.005</td>
<td align="right">0.001</td>
<td align="right">0.000</td>
<td align="right">0.011</td>
<td align="right">0.004</td>
<td align="right">0.013</td>
<td align="right">0.017</td>
<td align="right">0.002</td>
<td align="right">0.001</td>
<td align="right">0.001</td>
<td align="right">0.014</td>
<td align="right">0.002</td>
<td align="right">0.001</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">physeq_rel</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html" class="external-link">tax_table</a></span><span class="op">(</span><span class="op">)</span><span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<colgroup>
<col width="13%">
<col width="6%">
<col width="12%">
<col width="25%">
<col width="12%">
<col width="13%">
<col width="10%">
<col width="5%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="left">Kingdom</th>
<th align="left">Phylum</th>
<th align="left">Class</th>
<th align="left">Order</th>
<th align="left">Family</th>
<th align="left">Genus</th>
<th align="left">Species</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">0319-6G20</td>
<td align="left">Bacteria</td>
<td align="left">Proteobacteria</td>
<td align="left">Deltaproteobacteria</td>
<td align="left">Oligoflexales</td>
<td align="left">0319-6G20</td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">Kineosporia</td>
<td align="left">Bacteria</td>
<td align="left">Actinobacteria</td>
<td align="left">Actinobacteria(class)</td>
<td align="left">Kineosporiales</td>
<td align="left">Kineosporiaceae</td>
<td align="left">Kineosporia</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">OM1clade</td>
<td align="left">Bacteria</td>
<td align="left">Actinobacteria</td>
<td align="left">Acidimicrobiia</td>
<td align="left">Acidimicrobiales</td>
<td align="left">OM1clade</td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">Ca.Parcubacteria</td>
<td align="left">Bacteria</td>
<td align="left">Ca.Parcubacteria</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">Nitrosomonadaceae</td>
<td align="left">Bacteria</td>
<td align="left">Proteobacteria</td>
<td align="left">Betaproteobacteria</td>
<td align="left">Nitrosomonadales</td>
<td align="left">Nitrosomonadaceae</td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">Ca.Solibacter</td>
<td align="left">Bacteria</td>
<td align="left">Acidobacteria</td>
<td align="left">Solibacteres(Acidobacteriagroup3)</td>
<td align="left">Solibacterales</td>
<td align="left">Solibacteraceae</td>
<td align="left">Ca.Solibacter</td>
<td align="left"></td>
</tr>
</tbody>
</table>
<p>Check if everything looks good:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#|label: sample_sums</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_sums.html" class="external-link">sample_sums</a></span><span class="op">(</span><span class="va">physeq</span><span class="op">)</span></span>
<span><span class="co">#&gt;   TK20-2   TK20-3   TK20-4   TK20-5   TK20-6   TK20-7   TK20-8   TK20-9 </span></span>
<span><span class="co">#&gt;  5718973  4601365  3817254  3203439  2853679  8864332  2423598  5369228 </span></span>
<span><span class="co">#&gt;  TK20-10  TK20-11  TK20-12  TK20-13  TK20-14  TK20-15  TK20-16  TK20-17 </span></span>
<span><span class="co">#&gt;  2741181  3192877 10543326  3553206 21866346  2246072 22745242  5973567 </span></span>
<span><span class="co">#&gt;  TK20-18  TK20-19  TK20-20  TK20-21  TK20-22  TK20-23  TK20-24  TK20-25 </span></span>
<span><span class="co">#&gt;  3103575  2691406    23752  2153359  7644775  3084344  4072591  4390303 </span></span>
<span><span class="co">#&gt;  TK20-26  TK20-27  TK20-28  TK20-29  TK20-30 </span></span>
<span><span class="co">#&gt;  3966687  5019554    99428  3638618  2162107</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_names-methods.html" class="external-link">sample_names</a></span><span class="op">(</span><span class="va">physeq</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "TK20-2"  "TK20-3"  "TK20-4"  "TK20-5"  "TK20-6"  "TK20-7"  "TK20-8" </span></span>
<span><span class="co">#&gt;  [8] "TK20-9"  "TK20-10" "TK20-11" "TK20-12" "TK20-13" "TK20-14" "TK20-15"</span></span>
<span><span class="co">#&gt; [15] "TK20-16" "TK20-17" "TK20-18" "TK20-19" "TK20-20" "TK20-21" "TK20-22"</span></span>
<span><span class="co">#&gt; [22] "TK20-23" "TK20-24" "TK20-25" "TK20-26" "TK20-27" "TK20-28" "TK20-29"</span></span>
<span><span class="co">#&gt; [29] "TK20-30"</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/rank_names.html" class="external-link">rank_names</a></span><span class="op">(</span><span class="va">physeq</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Kingdom" "Phylum"  "Class"   "Order"   "Family"  "Genus"   "Species"</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_variables.html" class="external-link">sample_variables</a></span><span class="op">(</span><span class="va">physeq</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "layer" "age"   "pH"    "depth"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data-analysis">Data analysis<a class="anchor" aria-label="anchor" href="#data-analysis"></a>
</h2>
<div class="section level3">
<h3 id="perform-an-ordination-using-nonmetric-multidimensional-scaling">Perform an ordination using Nonmetric Multidimensional Scaling<a class="anchor" aria-label="anchor" href="#perform-an-ordination-using-nonmetric-multidimensional-scaling"></a>
</h3>
<p>Now, we ordinate using the NMDS method and bray distance. NMDS
performs a Non-metric MultiDimenstional Scaling of a sample-wise
ecological distance matrix onto a user-specified number of axes, k (in
this case 2).</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">physeq_nmds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/ordinate.html" class="external-link">ordinate</a></span><span class="op">(</span><span class="va">physeq_rel</span>, method <span class="op">=</span> <span class="st">"NMDS"</span>, distance <span class="op">=</span> <span class="st">"bray"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Wisconsin double standardization</span></span>
<span><span class="co">#&gt; Run 0 stress 0.0634 </span></span>
<span><span class="co">#&gt; Run 1 stress 0.0874 </span></span>
<span><span class="co">#&gt; Run 2 stress 0.0634 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 7.49e-06  max resid 2.03e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 3 stress 0.0634 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 9.79e-06  max resid 2.82e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 4 stress 0.0848 </span></span>
<span><span class="co">#&gt; Run 5 stress 0.0959 </span></span>
<span><span class="co">#&gt; Run 6 stress 0.0862 </span></span>
<span><span class="co">#&gt; Run 7 stress 0.0908 </span></span>
<span><span class="co">#&gt; Run 8 stress 0.0634 </span></span>
<span><span class="co">#&gt; ... New best solution</span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 6.77e-06  max resid 1.83e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 9 stress 0.0634 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 2.85e-05  max resid 7.63e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 10 stress 0.0634 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 1.83e-05  max resid 4.93e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 11 stress 0.0959 </span></span>
<span><span class="co">#&gt; Run 12 stress 0.0634 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 8.79e-06  max resid 2.35e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 13 stress 0.0634 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 1.07e-05  max resid 3.24e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 14 stress 0.0841 </span></span>
<span><span class="co">#&gt; Run 15 stress 0.0634 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 9.22e-06  max resid 2.3e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 16 stress 0.0634 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 2.32e-05  max resid 6.55e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 17 stress 0.0634 </span></span>
<span><span class="co">#&gt; ... New best solution</span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 8.81e-06  max resid 2.46e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 18 stress 0.0959 </span></span>
<span><span class="co">#&gt; Run 19 stress 0.0634 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 3.94e-06  max resid 1.11e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 20 stress 0.0908 </span></span>
<span><span class="co">#&gt; *** Best solution repeated 2 times</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Call:</span></span>
<span><span class="co">#&gt; metaMDS(comm = veganifyOTU(physeq), distance = distance) </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; global Multidimensional Scaling using monoMDS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Data:     wisconsin(veganifyOTU(physeq)) </span></span>
<span><span class="co">#&gt; Distance: bray </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Dimensions: 2 </span></span>
<span><span class="co">#&gt; Stress:     0.0634 </span></span>
<span><span class="co">#&gt; Stress type 1, weak ties</span></span>
<span><span class="co">#&gt; Best solution was repeated 2 times in 20 tries</span></span>
<span><span class="co">#&gt; The best solution was from try 17 (random start)</span></span>
<span><span class="co">#&gt; Scaling: centring, PC rotation, halfchange scaling </span></span>
<span><span class="co">#&gt; Species: expanded scores based on 'wisconsin(veganifyOTU(physeq))'</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="goodness-of-fit-and-shepard-plot-for-nonmetric-multidimensional-scaling">Goodness of Fit and Shepard Plot for Nonmetric Multidimensional
Scaling<a class="anchor" aria-label="anchor" href="#goodness-of-fit-and-shepard-plot-for-nonmetric-multidimensional-scaling"></a>
</h3>
<p>Now, we find the goodness of fit measure for the points in the
previous nonmetric multidimensional scaling. The Shepard diagram is
shown in the following figure:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">vegan</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/vegan/man/goodness.metaMDS.html" class="external-link">stressplot</a></span><span class="op">(</span><span class="va">physeq_nmds</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="phyloseq_analysis_files/figure-html/fig-stressplot-1.png" alt="A plot of ordination distances and monotone or linear fit line against original dissimilarities." width="680.314960629921"><p class="caption">
A plot of ordination distances and monotone or linear fit line against
original dissimilarities.
</p>
</div>
<p>Now, we’ll create a function for a) ordinate and b) plot. This way
we’ll be able to plot for different subsets of taxa at the same
time.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot_nmds</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">y</span> <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/ordinate.html" class="external-link">ordinate</a></span><span class="op">(</span><span class="va">x</span>, method <span class="op">=</span> <span class="st">"NMDS"</span>, distance <span class="op">=</span> <span class="st">"bray"</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/plot_ordination.html" class="external-link">plot_ordination</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span>, <span class="va">...</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>    legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"right"</span></span>
<span>  <span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/stat_ellipse.html" class="external-link">stat_ellipse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">nmds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  all <span class="op">=</span> <span class="fu">plot_nmds</span><span class="op">(</span></span>
<span>    <span class="va">physeq_rel</span>,type <span class="op">=</span><span class="st">"samples"</span>, color <span class="op">=</span> <span class="st">"layer"</span>,shape <span class="op">=</span> <span class="st">"layer"</span></span>
<span>    <span class="op">)</span>,</span>
<span>  phyla <span class="op">=</span> <span class="va">physeq_rel</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_glom.html" class="external-link">tax_glom</a></span><span class="op">(</span>taxrank <span class="op">=</span> <span class="st">"Phylum"</span>, NArm <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">plot_nmds</span><span class="op">(</span></span>
<span>      type <span class="op">=</span><span class="st">"samples"</span>, color <span class="op">=</span> <span class="st">"layer"</span>, shape <span class="op">=</span> <span class="st">"layer"</span></span>
<span>      <span class="op">)</span>,</span>
<span>  class <span class="op">=</span> <span class="va">physeq_rel</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_glom.html" class="external-link">tax_glom</a></span><span class="op">(</span>taxrank <span class="op">=</span> <span class="st">"Class"</span>, NArm <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">plot_nmds</span><span class="op">(</span></span>
<span>      type <span class="op">=</span><span class="st">"samples"</span>, color <span class="op">=</span> <span class="st">"layer"</span>, shape <span class="op">=</span> <span class="st">"layer"</span></span>
<span>      <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Wisconsin double standardization</span></span>
<span><span class="co">#&gt; Run 0 stress 0.0634 </span></span>
<span><span class="co">#&gt; Run 1 stress 0.0634 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 4.92e-06  max resid 9.9e-06 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 2 stress 0.0634 </span></span>
<span><span class="co">#&gt; ... New best solution</span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 3.07e-06  max resid 6.75e-06 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 3 stress 0.0634 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 3.2e-05  max resid 8.94e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 4 stress 0.0634 </span></span>
<span><span class="co">#&gt; ... New best solution</span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 8.12e-06  max resid 2.18e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 5 stress 0.0918 </span></span>
<span><span class="co">#&gt; Run 6 stress 0.0848 </span></span>
<span><span class="co">#&gt; Run 7 stress 0.0862 </span></span>
<span><span class="co">#&gt; Run 8 stress 0.0821 </span></span>
<span><span class="co">#&gt; Run 9 stress 0.0874 </span></span>
<span><span class="co">#&gt; Run 10 stress 0.0634 </span></span>
<span><span class="co">#&gt; ... New best solution</span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 1.05e-05  max resid 2.52e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 11 stress 0.0634 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 3.41e-05  max resid 9.33e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 12 stress 0.0634 </span></span>
<span><span class="co">#&gt; ... New best solution</span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 6.59e-06  max resid 1.81e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 13 stress 0.0634 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 5.27e-05  max resid 0.00013 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 14 stress 0.0898 </span></span>
<span><span class="co">#&gt; Run 15 stress 0.0848 </span></span>
<span><span class="co">#&gt; Run 16 stress 0.0634 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 2.19e-05  max resid 5.9e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 17 stress 0.0848 </span></span>
<span><span class="co">#&gt; Run 18 stress 0.0634 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 2.43e-05  max resid 6.65e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 19 stress 0.0634 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 6.53e-05  max resid 0.000161 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 20 stress 0.0908 </span></span>
<span><span class="co">#&gt; *** Best solution repeated 5 times</span></span>
<span><span class="co">#&gt; Square root transformation</span></span>
<span><span class="co">#&gt; Wisconsin double standardization</span></span>
<span><span class="co">#&gt; Run 0 stress 0.0653 </span></span>
<span><span class="co">#&gt; Run 1 stress 0.0653 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 6.68e-06  max resid 2.44e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 2 stress 0.122 </span></span>
<span><span class="co">#&gt; Run 3 stress 0.0653 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 1.01e-05  max resid 4.19e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 4 stress 0.0653 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 4.17e-06  max resid 1.63e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 5 stress 0.122 </span></span>
<span><span class="co">#&gt; Run 6 stress 0.107 </span></span>
<span><span class="co">#&gt; Run 7 stress 0.0874 </span></span>
<span><span class="co">#&gt; Run 8 stress 0.0874 </span></span>
<span><span class="co">#&gt; Run 9 stress 0.107 </span></span>
<span><span class="co">#&gt; Run 10 stress 0.106 </span></span>
<span><span class="co">#&gt; Run 11 stress 0.0653 </span></span>
<span><span class="co">#&gt; ... New best solution</span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 1.73e-06  max resid 6.71e-06 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 12 stress 0.112 </span></span>
<span><span class="co">#&gt; Run 13 stress 0.0653 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 3.06e-05  max resid 0.000135 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 14 stress 0.106 </span></span>
<span><span class="co">#&gt; Run 15 stress 0.0653 </span></span>
<span><span class="co">#&gt; ... New best solution</span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 2.66e-06  max resid 1.1e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 16 stress 0.0874 </span></span>
<span><span class="co">#&gt; Run 17 stress 0.0653 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 5.43e-06  max resid 2.27e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 18 stress 0.0653 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 1.8e-06  max resid 5.24e-06 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 19 stress 0.0874 </span></span>
<span><span class="co">#&gt; Run 20 stress 0.0653 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 5.93e-06  max resid 2.59e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; *** Best solution repeated 4 times</span></span>
<span><span class="co">#&gt; Square root transformation</span></span>
<span><span class="co">#&gt; Wisconsin double standardization</span></span>
<span><span class="co">#&gt; Run 0 stress 0.0983 </span></span>
<span><span class="co">#&gt; Run 1 stress 0.1 </span></span>
<span><span class="co">#&gt; Run 2 stress 0.087 </span></span>
<span><span class="co">#&gt; ... New best solution</span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 0.0748  max resid 0.243 </span></span>
<span><span class="co">#&gt; Run 3 stress 0.105 </span></span>
<span><span class="co">#&gt; Run 4 stress 0.384 </span></span>
<span><span class="co">#&gt; Run 5 stress 0.0865 </span></span>
<span><span class="co">#&gt; ... New best solution</span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 0.0246  max resid 0.118 </span></span>
<span><span class="co">#&gt; Run 6 stress 0.0887 </span></span>
<span><span class="co">#&gt; Run 7 stress 0.0717 </span></span>
<span><span class="co">#&gt; ... New best solution</span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 0.102  max resid 0.359 </span></span>
<span><span class="co">#&gt; Run 8 stress 0.103 </span></span>
<span><span class="co">#&gt; Run 9 stress 0.0717 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 3.88e-05  max resid 0.00013 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 10 stress 0.0865 </span></span>
<span><span class="co">#&gt; Run 11 stress 0.104 </span></span>
<span><span class="co">#&gt; Run 12 stress 0.0717 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 4.27e-05  max resid 0.00015 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 13 stress 0.1 </span></span>
<span><span class="co">#&gt; Run 14 stress 0.0717 </span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 0.00526  max resid 0.0225 </span></span>
<span><span class="co">#&gt; Run 15 stress 0.105 </span></span>
<span><span class="co">#&gt; Run 16 stress 0.085 </span></span>
<span><span class="co">#&gt; Run 17 stress 0.0717 </span></span>
<span><span class="co">#&gt; ... New best solution</span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 1.75e-05  max resid 6.09e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; Run 18 stress 0.101 </span></span>
<span><span class="co">#&gt; Run 19 stress 0.0865 </span></span>
<span><span class="co">#&gt; Run 20 stress 0.0717 </span></span>
<span><span class="co">#&gt; ... New best solution</span></span>
<span><span class="co">#&gt; ... Procrustes: rmse 1.86e-05  max resid 6.83e-05 </span></span>
<span><span class="co">#&gt; ... Similar to previous best</span></span>
<span><span class="co">#&gt; *** Best solution repeated 1 times</span></span></code></pre></div>
<p>You can plot each plot individually:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nmds</span><span class="op">$</span><span class="va">all</span></span></code></pre></div>
<p>Or use plot_grid for a list of plots:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span>plotlist <span class="op">=</span> <span class="va">nmds</span><span class="op">)</span></span></code></pre></div>
<p><img src="phyloseq_analysis_files/figure-html/unnamed-chunk-21-1.png" width="680.314960629921"></p>
</div>
<div class="section level3">
<h3 id="richness-plot">Richness plot<a class="anchor" aria-label="anchor" href="#richness-plot"></a>
</h3>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">###----Richness all----</span></span>
<span></span>
<span><span class="va">richness_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/plot_richness.html" class="external-link">plot_richness</a></span><span class="op">(</span></span>
<span>  <span class="va">physeq</span>, x<span class="op">=</span><span class="st">"layer"</span>, color <span class="op">=</span>  <span class="st">"depth"</span>,</span>
<span>  measures<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Observed"</span>,<span class="st">"Shannon"</span>,<span class="st">"Simpson"</span>,<span class="st">"InvSimpson"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">richness_plot</span> <span class="op">&lt;-</span> <span class="va">richness_plot</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">richness_plot</span><span class="op">$</span><span class="va">data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>color <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span>, alpha <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>    legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"right"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">richness_plot</span></span></code></pre></div>
<p><img src="phyloseq_analysis_files/figure-html/unnamed-chunk-22-1.png" width="680.314960629921"></p>
</div>
<div class="section level3">
<h3 id="stacked-bar-plots">Stacked bar plots<a class="anchor" aria-label="anchor" href="#stacked-bar-plots"></a>
</h3>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">physeq_mean_rel_phylum</span> <span class="op">&lt;-</span> <span class="va">physeq_mean_rel</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_glom.html" class="external-link">tax_glom</a></span><span class="op">(</span>taxrank <span class="op">=</span> <span class="st">"Phylum"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transformcounts.html" class="external-link">transform_sample_counts</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> </span></code></pre></div>
<p>Since we get too many phyla to plot in a stacked barplot, we will
filter the low abundant ones and put them into one category. To do this,
we will use the tidyverse again. First, we will create a normal data
frame out of the phyloseq object and then add another column where all
taxa with abundance lower than 3% will be renamed to “&lt; 3%”.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">physeq_mean_rel_phylumDF</span> <span class="op">&lt;-</span> <span class="va">physeq_mean_rel_phylum</span> <span class="op">|&gt;</span></span>
<span>  <span class="co">#transform phyloseq object to a data frame (DF)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/psmelt.html" class="external-link">psmelt</a></span><span class="op">(</span><span class="op">)</span><span class="op">|&gt;</span></span>
<span>  <span class="co">#make the phyla characters, not factors</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Phylum <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">Phylum</span><span class="op">)</span><span class="op">)</span><span class="op">|&gt;</span></span>
<span>  <span class="co">#there are some reads that were assigned only to the kingdom level, </span></span>
<span>  <span class="co"># i.e. NA on the phylum level, so we will rename them</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>    Phylum <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/replace.html" class="external-link">replace</a></span><span class="op">(</span><span class="va">Phylum</span>, <span class="va">Phylum</span> <span class="op">==</span> <span class="st">"NA"</span>, <span class="st">"unassigned"</span><span class="op">)</span>,</span>
<span>    Phylum2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/replace.html" class="external-link">replace</a></span><span class="op">(</span><span class="va">Phylum</span>, <span class="va">Abundance</span> <span class="op">&lt;</span> <span class="fl">3</span>, <span class="st">"&lt; 3%"</span><span class="op">)</span></span>
<span>    <span class="op">)</span><span class="op">|&gt;</span></span>
<span>  <span class="co">#reorder the phyla so that they are stacked according to abundance</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>    Phylum2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/reorder.factor.html" class="external-link">reorder</a></span><span class="op">(</span><span class="va">Phylum2</span>, <span class="va">Abundance</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>Now, we plot the stacked plot. You can change colors editing function
get_wants_hue. Default is optimized for colorblindness.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n_phyla</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">physeq_mean_rel_phylumDF</span><span class="op">$</span><span class="va">Phylum2</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">stacked_plot</span> <span class="op">&lt;-</span> <span class="va">physeq_mean_rel_phylumDF</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">depth</span>, <span class="va">Abundance</span>, fill<span class="op">=</span><span class="va">Phylum2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x<span class="op">=</span> <span class="st">"depth [cm]"</span>,y<span class="op">=</span> <span class="st">"Relative abundance [%]"</span>,</span>
<span>     fill<span class="op">=</span> <span class="st">"Phyla"</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="../reference/get_wants_hue.html">get_wants_hue</a></span><span class="op">(</span><span class="va">n_phyla</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position<span class="op">=</span><span class="st">"bottom"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_continuous</a></span><span class="op">(</span>breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">100</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">stacked_plot</span> </span></code></pre></div>
<p><img src="phyloseq_analysis_files/figure-html/unnamed-chunk-25-1.png" width="680.314960629921"></p>
<p>You can obtain figures with more than one plot using plot_grid. For
example:</p>
<p><a href="https://wilkelab.org/cowplot/articles/plot_grid.html" class="external-link uri">https://wilkelab.org/cowplot/articles/plot_grid.html</a></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">final</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span></span>
<span>  <span class="va">stacked_plot</span>, <span class="va">richness_plot</span>,nrow <span class="op">=</span> <span class="fl">2</span>,labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"B"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">final</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="anova">ANOVA<a class="anchor" aria-label="anchor" href="#anova"></a>
</h3>
<p>First, create objects for microbiome package:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">otu_microbiome</span> <span class="op">&lt;-</span> <span class="fu">microbiome</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microbiome/man/abundances.html" class="external-link">abundances</a></span><span class="op">(</span><span class="va">physeq</span><span class="op">)</span></span>
<span><span class="va">meta_microbiome</span> <span class="op">&lt;-</span> <span class="fu">microbiome</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microbiome/man/meta.html" class="external-link">meta</a></span><span class="op">(</span><span class="va">physeq</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">otu_microbiome</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="7%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">TK20-2</th>
<th align="right">TK20-3</th>
<th align="right">TK20-4</th>
<th align="right">TK20-5</th>
<th align="right">TK20-6</th>
<th align="right">TK20-7</th>
<th align="right">TK20-8</th>
<th align="right">TK20-9</th>
<th align="right">TK20-10</th>
<th align="right">TK20-11</th>
<th align="right">TK20-12</th>
<th align="right">TK20-13</th>
<th align="right">TK20-14</th>
<th align="right">TK20-15</th>
<th align="right">TK20-16</th>
<th align="right">TK20-17</th>
<th align="right">TK20-18</th>
<th align="right">TK20-19</th>
<th align="right">TK20-20</th>
<th align="right">TK20-21</th>
<th align="right">TK20-22</th>
<th align="right">TK20-23</th>
<th align="right">TK20-24</th>
<th align="right">TK20-25</th>
<th align="right">TK20-26</th>
<th align="right">TK20-27</th>
<th align="right">TK20-28</th>
<th align="right">TK20-29</th>
<th align="right">TK20-30</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">0319-6G20</td>
<td align="right">344</td>
<td align="right">338</td>
<td align="right">872</td>
<td align="right">944</td>
<td align="right">187</td>
<td align="right">83</td>
<td align="right">17</td>
<td align="right">47</td>
<td align="right">6</td>
<td align="right">412</td>
<td align="right">774</td>
<td align="right">355</td>
<td align="right">4769</td>
<td align="right">992</td>
<td align="right">2959</td>
<td align="right">18</td>
<td align="right">1</td>
<td align="right">30</td>
<td align="right">0</td>
<td align="right">139</td>
<td align="right">646</td>
<td align="right">363</td>
<td align="right">962</td>
<td align="right">956</td>
<td align="right">884</td>
<td align="right">246</td>
<td align="right">0</td>
<td align="right">23</td>
<td align="right">70</td>
</tr>
<tr class="even">
<td align="left">Kineosporia</td>
<td align="right">5776</td>
<td align="right">14725</td>
<td align="right">4922</td>
<td align="right">3000</td>
<td align="right">3304</td>
<td align="right">646</td>
<td align="right">298</td>
<td align="right">448</td>
<td align="right">401</td>
<td align="right">10733</td>
<td align="right">5590</td>
<td align="right">9648</td>
<td align="right">80899</td>
<td align="right">8926</td>
<td align="right">783</td>
<td align="right">234</td>
<td align="right">65</td>
<td align="right">190</td>
<td align="right">71</td>
<td align="right">13144</td>
<td align="right">6740</td>
<td align="right">9200</td>
<td align="right">4332</td>
<td align="right">4624</td>
<td align="right">911</td>
<td align="right">186</td>
<td align="right">4</td>
<td align="right">149</td>
<td align="right">412</td>
</tr>
<tr class="odd">
<td align="left">OM1clade</td>
<td align="right">4</td>
<td align="right">5</td>
<td align="right">3</td>
<td align="right">2</td>
<td align="right">9</td>
<td align="right">295</td>
<td align="right">4380</td>
<td align="right">10308</td>
<td align="right">7</td>
<td align="right">16</td>
<td align="right">13</td>
<td align="right">4</td>
<td align="right">29</td>
<td align="right">0</td>
<td align="right">23</td>
<td align="right">167</td>
<td align="right">1190</td>
<td align="right">5087</td>
<td align="right">3</td>
<td align="right">3</td>
<td align="right">15</td>
<td align="right">11</td>
<td align="right">10</td>
<td align="right">9</td>
<td align="right">1</td>
<td align="right">5</td>
<td align="right">5</td>
<td align="right">8557</td>
<td align="right">16</td>
</tr>
<tr class="even">
<td align="left">Ca.Parcubacteria</td>
<td align="right">83</td>
<td align="right">73</td>
<td align="right">56</td>
<td align="right">11</td>
<td align="right">18</td>
<td align="right">64</td>
<td align="right">125</td>
<td align="right">306</td>
<td align="right">1</td>
<td align="right">84</td>
<td align="right">181</td>
<td align="right">61</td>
<td align="right">387</td>
<td align="right">4</td>
<td align="right">83</td>
<td align="right">42</td>
<td align="right">23</td>
<td align="right">156</td>
<td align="right">0</td>
<td align="right">36</td>
<td align="right">117</td>
<td align="right">52</td>
<td align="right">113</td>
<td align="right">25</td>
<td align="right">16</td>
<td align="right">63</td>
<td align="right">2</td>
<td align="right">43</td>
<td align="right">9</td>
</tr>
<tr class="odd">
<td align="left">Nitrosomonadaceae</td>
<td align="right">1798</td>
<td align="right">886</td>
<td align="right">2251</td>
<td align="right">3176</td>
<td align="right">437</td>
<td align="right">1598</td>
<td align="right">2589</td>
<td align="right">7291</td>
<td align="right">4790</td>
<td align="right">261</td>
<td align="right">3532</td>
<td align="right">785</td>
<td align="right">14943</td>
<td align="right">1829</td>
<td align="right">8676</td>
<td align="right">1151</td>
<td align="right">2294</td>
<td align="right">3138</td>
<td align="right">8</td>
<td align="right">174</td>
<td align="right">2186</td>
<td align="right">555</td>
<td align="right">2508</td>
<td align="right">2096</td>
<td align="right">3313</td>
<td align="right">2074</td>
<td align="right">65</td>
<td align="right">2503</td>
<td align="right">2538</td>
</tr>
<tr class="even">
<td align="left">Ca.Solibacter</td>
<td align="right">292</td>
<td align="right">595</td>
<td align="right">540</td>
<td align="right">29</td>
<td align="right">89</td>
<td align="right">75</td>
<td align="right">12</td>
<td align="right">63</td>
<td align="right">11</td>
<td align="right">234</td>
<td align="right">388</td>
<td align="right">362</td>
<td align="right">2452</td>
<td align="right">18</td>
<td align="right">254</td>
<td align="right">28</td>
<td align="right">143</td>
<td align="right">31</td>
<td align="right">0</td>
<td align="right">231</td>
<td align="right">318</td>
<td align="right">414</td>
<td align="right">708</td>
<td align="right">74</td>
<td align="right">25</td>
<td align="right">38</td>
<td align="right">14</td>
<td align="right">67</td>
<td align="right">27</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meta_microbiome</span> <span class="op">|&gt;</span> <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="left">layer</th>
<th align="left">age</th>
<th align="right">pH</th>
<th align="right">depth</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">TK20-2</td>
<td align="left">AL</td>
<td align="left">AY</td>
<td align="right">4.02</td>
<td align="right">20</td>
</tr>
<tr class="even">
<td align="left">TK20-3</td>
<td align="left">AL</td>
<td align="left">AY</td>
<td align="right">4.29</td>
<td align="right">30</td>
</tr>
<tr class="odd">
<td align="left">TK20-4</td>
<td align="left">AL</td>
<td align="left">AY</td>
<td align="right">4.25</td>
<td align="right">40</td>
</tr>
<tr class="even">
<td align="left">TK20-5</td>
<td align="left">TZ1</td>
<td align="left">AM</td>
<td align="right">4.64</td>
<td align="right">50</td>
</tr>
<tr class="odd">
<td align="left">TK20-6</td>
<td align="left">TZ1</td>
<td align="left">AM</td>
<td align="right">4.13</td>
<td align="right">60</td>
</tr>
<tr class="even">
<td align="left">TK20-7</td>
<td align="left">TZ1</td>
<td align="left">AO</td>
<td align="right">4.86</td>
<td align="right">70</td>
</tr>
<tr class="odd">
<td align="left">TK20-8</td>
<td align="left">TZ2</td>
<td align="left">AO</td>
<td align="right">4.48</td>
<td align="right">80</td>
</tr>
<tr class="even">
<td align="left">TK20-9</td>
<td align="left">TZ2</td>
<td align="left">AO</td>
<td align="right">4.57</td>
<td align="right">90</td>
</tr>
<tr class="odd">
<td align="left">TK20-10</td>
<td align="left">PF</td>
<td align="left">AO</td>
<td align="right">4.91</td>
<td align="right">100</td>
</tr>
<tr class="even">
<td align="left">TK20-11</td>
<td align="left">AL</td>
<td align="left">AY</td>
<td align="right">4.22</td>
<td align="right">10</td>
</tr>
<tr class="odd">
<td align="left">TK20-12</td>
<td align="left">AL</td>
<td align="left">AY</td>
<td align="right">4.02</td>
<td align="right">20</td>
</tr>
<tr class="even">
<td align="left">TK20-13</td>
<td align="left">AL</td>
<td align="left">AY</td>
<td align="right">4.29</td>
<td align="right">30</td>
</tr>
<tr class="odd">
<td align="left">TK20-14</td>
<td align="left">AL</td>
<td align="left">AY</td>
<td align="right">4.25</td>
<td align="right">40</td>
</tr>
<tr class="even">
<td align="left">TK20-15</td>
<td align="left">TZ1</td>
<td align="left">AM</td>
<td align="right">4.64</td>
<td align="right">50</td>
</tr>
<tr class="odd">
<td align="left">TK20-16</td>
<td align="left">TZ1</td>
<td align="left">AM</td>
<td align="right">4.13</td>
<td align="right">60</td>
</tr>
<tr class="even">
<td align="left">TK20-17</td>
<td align="left">TZ1</td>
<td align="left">AO</td>
<td align="right">4.86</td>
<td align="right">70</td>
</tr>
<tr class="odd">
<td align="left">TK20-18</td>
<td align="left">TZ2</td>
<td align="left">AO</td>
<td align="right">4.48</td>
<td align="right">80</td>
</tr>
<tr class="even">
<td align="left">TK20-19</td>
<td align="left">TZ2</td>
<td align="left">AO</td>
<td align="right">4.57</td>
<td align="right">90</td>
</tr>
<tr class="odd">
<td align="left">TK20-20</td>
<td align="left">PF</td>
<td align="left">AO</td>
<td align="right">4.91</td>
<td align="right">100</td>
</tr>
<tr class="even">
<td align="left">TK20-21</td>
<td align="left">AL</td>
<td align="left">AY</td>
<td align="right">4.22</td>
<td align="right">10</td>
</tr>
<tr class="odd">
<td align="left">TK20-22</td>
<td align="left">AL</td>
<td align="left">AY</td>
<td align="right">4.02</td>
<td align="right">20</td>
</tr>
<tr class="even">
<td align="left">TK20-23</td>
<td align="left">AL</td>
<td align="left">AY</td>
<td align="right">4.29</td>
<td align="right">30</td>
</tr>
<tr class="odd">
<td align="left">TK20-24</td>
<td align="left">AL</td>
<td align="left">AY</td>
<td align="right">4.25</td>
<td align="right">40</td>
</tr>
<tr class="even">
<td align="left">TK20-25</td>
<td align="left">TZ1</td>
<td align="left">AM</td>
<td align="right">4.64</td>
<td align="right">50</td>
</tr>
<tr class="odd">
<td align="left">TK20-26</td>
<td align="left">TZ1</td>
<td align="left">AM</td>
<td align="right">4.13</td>
<td align="right">60</td>
</tr>
<tr class="even">
<td align="left">TK20-27</td>
<td align="left">TZ1</td>
<td align="left">AO</td>
<td align="right">4.86</td>
<td align="right">70</td>
</tr>
<tr class="odd">
<td align="left">TK20-28</td>
<td align="left">TZ2</td>
<td align="left">AO</td>
<td align="right">4.48</td>
<td align="right">80</td>
</tr>
<tr class="even">
<td align="left">TK20-29</td>
<td align="left">TZ2</td>
<td align="left">AO</td>
<td align="right">4.57</td>
<td align="right">90</td>
</tr>
<tr class="odd">
<td align="left">TK20-30</td>
<td align="left">PF</td>
<td align="left">AO</td>
<td align="right">4.91</td>
<td align="right">100</td>
</tr>
</tbody>
</table>
<p>Now, we use adonis for those variables we are interested. We should
probably use adonis2, because adonis have been deprecated.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">permanova_depth</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/adonis.html" class="external-link">adonis2</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">otu_microbiome</span><span class="op">)</span><span class="op">~</span><span class="va">depth</span>,</span>
<span>  data <span class="op">=</span> <span class="va">meta_microbiome</span>,</span>
<span>  permutations<span class="op">=</span><span class="fl">999</span>, method <span class="op">=</span> <span class="st">"bray"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">permanova_depth</span></span>
<span><span class="co">#&gt; Permutation test for adonis under reduced model</span></span>
<span><span class="co">#&gt; Terms added sequentially (first to last)</span></span>
<span><span class="co">#&gt; Permutation: free</span></span>
<span><span class="co">#&gt; Number of permutations: 999</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; adonis2(formula = t(otu_microbiome) ~ depth, data = meta_microbiome, permutations = 999, method = "bray")</span></span>
<span><span class="co">#&gt;          Df SumOfSqs    R2    F Pr(&gt;F)    </span></span>
<span><span class="co">#&gt; depth     1     1.91 0.255 9.26  0.001 ***</span></span>
<span><span class="co">#&gt; Residual 27     5.56 0.745                </span></span>
<span><span class="co">#&gt; Total    28     7.46 1.000                </span></span>
<span><span class="co">#&gt; ---</span></span>
<span><span class="co">#&gt; Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span></code></pre></div>
<p>Now layer:</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">permanova_layer</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/adonis.html" class="external-link">adonis2</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">otu_microbiome</span><span class="op">)</span><span class="op">~</span><span class="va">layer</span>,</span>
<span>  data <span class="op">=</span> <span class="va">meta_microbiome</span>,</span>
<span>  permutations<span class="op">=</span><span class="fl">999</span>, method <span class="op">=</span> <span class="st">"bray"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">permanova_layer</span></span>
<span><span class="co">#&gt; Permutation test for adonis under reduced model</span></span>
<span><span class="co">#&gt; Terms added sequentially (first to last)</span></span>
<span><span class="co">#&gt; Permutation: free</span></span>
<span><span class="co">#&gt; Number of permutations: 999</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; adonis2(formula = t(otu_microbiome) ~ layer, data = meta_microbiome, permutations = 999, method = "bray")</span></span>
<span><span class="co">#&gt;          Df SumOfSqs    R2    F Pr(&gt;F)    </span></span>
<span><span class="co">#&gt; layer     3     3.17 0.425 6.15  0.001 ***</span></span>
<span><span class="co">#&gt; Residual 25     4.30 0.575                </span></span>
<span><span class="co">#&gt; Total    28     7.46 1.000                </span></span>
<span><span class="co">#&gt; ---</span></span>
<span><span class="co">#&gt; Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span></code></pre></div>
<p>If you have several “models”, consider a more general approach using
lists:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">adonis2_formulas</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  layer <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">otu_microbiome</span><span class="op">)</span><span class="op">~</span><span class="va">layer</span>,</span>
<span>  depth <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">otu_microbiome</span><span class="op">)</span><span class="op">~</span><span class="va">depth</span></span>
<span><span class="op">)</span></span>
<span><span class="va">permanovas</span> <span class="op">&lt;-</span> <span class="va">adonis2_formulas</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span></span>
<span>    <span class="va">adonis2</span>, data <span class="op">=</span> <span class="va">meta_microbiome</span>,</span>
<span>    permutations<span class="op">=</span><span class="fl">999</span>, method <span class="op">=</span> <span class="st">"bray"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="va">permanovas</span><span class="op">$</span><span class="va">layer</span></span>
<span><span class="co">#&gt; Permutation test for adonis under reduced model</span></span>
<span><span class="co">#&gt; Terms added sequentially (first to last)</span></span>
<span><span class="co">#&gt; Permutation: free</span></span>
<span><span class="co">#&gt; Number of permutations: 999</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; .f(formula = .x[[i]], data = ..1, permutations = 999, method = "bray")</span></span>
<span><span class="co">#&gt;          Df SumOfSqs    R2    F Pr(&gt;F)    </span></span>
<span><span class="co">#&gt; layer     3     3.17 0.425 6.15  0.001 ***</span></span>
<span><span class="co">#&gt; Residual 25     4.30 0.575                </span></span>
<span><span class="co">#&gt; Total    28     7.46 1.000                </span></span>
<span><span class="co">#&gt; ---</span></span>
<span><span class="co">#&gt; Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="va">permanovas</span><span class="op">$</span><span class="va">depth</span></span>
<span><span class="co">#&gt; Permutation test for adonis under reduced model</span></span>
<span><span class="co">#&gt; Terms added sequentially (first to last)</span></span>
<span><span class="co">#&gt; Permutation: free</span></span>
<span><span class="co">#&gt; Number of permutations: 999</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; .f(formula = .x[[i]], data = ..1, permutations = 999, method = "bray")</span></span>
<span><span class="co">#&gt;          Df SumOfSqs    R2    F Pr(&gt;F)    </span></span>
<span><span class="co">#&gt; depth     1     1.91 0.255 9.26  0.001 ***</span></span>
<span><span class="co">#&gt; Residual 27     5.56 0.745                </span></span>
<span><span class="co">#&gt; Total    28     7.46 1.000                </span></span>
<span><span class="co">#&gt; ---</span></span>
<span><span class="co">#&gt; Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="multivariate-homogeneity-of-groups-dispersions">Multivariate homogeneity of groups dispersions<a class="anchor" aria-label="anchor" href="#multivariate-homogeneity-of-groups-dispersions"></a>
</h3>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate distance</span></span>
<span><span class="va">dist</span> <span class="op">&lt;-</span> <span class="fu">vegan</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/vegan/man/vegdist.html" class="external-link">vegdist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">otu_microbiome</span><span class="op">)</span>, method<span class="op">=</span><span class="st">"bray"</span><span class="op">)</span></span>
<span><span class="co"># betadisper is a multivariate analogue of Levene's test for homogeneity of variances</span></span>
<span><span class="va">mod_layer</span> <span class="op">&lt;-</span> <span class="fu">vegan</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/vegan/man/betadisper.html" class="external-link">betadisper</a></span><span class="op">(</span></span>
<span>  <span class="va">dist</span>, <span class="va">meta_microbiome</span><span class="op">$</span><span class="va">layer</span>, type<span class="op">=</span><span class="st">"centroid"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">mod_layer</span><span class="op">$</span><span class="va">layer</span> <span class="op">&lt;-</span> <span class="va">meta_microbiome</span><span class="op">$</span><span class="va">layer</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="tukeyhsd">TukeyHSD<a class="anchor" aria-label="anchor" href="#tukeyhsd"></a>
</h3>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">tukey_test</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/TukeyHSD.html" class="external-link">TukeyHSD</a></span><span class="op">(</span><span class="va">mod_layer</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Tukey multiple comparisons of means</span></span>
<span><span class="co">#&gt;     95% family-wise confidence level</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Fit: aov(formula = distances ~ group, data = df)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $group</span></span>
<span><span class="co">#&gt;             diff    lwr   upr p adj</span></span>
<span><span class="co">#&gt; PF-AL    0.12407 -0.111 0.359 0.480</span></span>
<span><span class="co">#&gt; TZ1-AL   0.05798 -0.104 0.220 0.760</span></span>
<span><span class="co">#&gt; TZ2-AL   0.05981 -0.123 0.243 0.806</span></span>
<span><span class="co">#&gt; TZ1-PF  -0.06609 -0.307 0.174 0.873</span></span>
<span><span class="co">#&gt; TZ2-PF  -0.06426 -0.319 0.191 0.899</span></span>
<span><span class="co">#&gt; TZ2-TZ1  0.00183 -0.188 0.192 1.000</span></span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">tukey_test</span><span class="op">)</span></span></code></pre></div>
<p><img src="phyloseq_analysis_files/figure-html/unnamed-chunk-33-1.png" width="680.314960629921"></p>
</div>
<div class="section level3">
<h3 id="pcoa-with-variance-in-axis">PCoA with variance in axis<a class="anchor" aria-label="anchor" href="#pcoa-with-variance-in-axis"></a>
</h3>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">get_variance</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">mod</span>, <span class="va">number</span>, <span class="va">digits</span> <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">variance</span> <span class="op">&lt;-</span> <span class="va">mod</span><span class="op">$</span><span class="va">eig</span><span class="op">[</span><span class="va">number</span><span class="op">]</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">mod</span><span class="op">$</span><span class="va">eig</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span></span>
<span>  <span class="va">variance</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">digits</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/format.html" class="external-link">format</a></span><span class="op">(</span>nsmall <span class="op">=</span> <span class="va">digits</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">x_label</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">'PCoA 1 ('</span>, <span class="fu">get_variance</span><span class="op">(</span><span class="va">mod_layer</span>, <span class="fl">1</span><span class="op">)</span>,<span class="st">'%)'</span><span class="op">)</span></span>
<span><span class="va">y_label</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">'PCoA 2 ('</span>, <span class="fu">get_variance</span><span class="op">(</span><span class="va">mod_layer</span>, <span class="fl">2</span><span class="op">)</span>,<span class="st">'%)'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span></span>
<span>  <span class="va">mod_layer</span>,  hull<span class="op">=</span><span class="cn">FALSE</span>, ellipse<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>  main <span class="op">=</span> <span class="st">"PCoA, layer"</span>, sub<span class="op">=</span><span class="cn">NULL</span>,</span>
<span>  col<span class="op">=</span> <span class="fu">RauENVS</span><span class="fu">::</span><span class="fu"><a href="../reference/get_wants_hue.html">get_wants_hue</a></span><span class="op">(</span>seed <span class="op">=</span> <span class="fl">10</span>, n <span class="op">=</span> <span class="fl">3</span>, palette <span class="op">=</span> <span class="st">"default"</span><span class="op">)</span>,</span>
<span>  cex<span class="op">=</span><span class="fl">2</span>, lwd<span class="op">=</span><span class="fl">1</span>,</span>
<span>  xlab <span class="op">=</span> <span class="va">x_label</span>,</span>
<span>  ylab <span class="op">=</span> <span class="va">y_label</span></span>
<span>  <span class="op">)</span> <span class="co">#+</span></span></code></pre></div>
<p><img src="phyloseq_analysis_files/figure-html/unnamed-chunk-34-1.png" width="680.314960629921"></p>
</div>
<div class="section level3">
<h3 id="boxplot-of-distances-with-pvalues-from-tukey-test">Boxplot of distances with pvalues from Tukey Test<a class="anchor" aria-label="anchor" href="#boxplot-of-distances-with-pvalues-from-tukey-test"></a>
</h3>
<p>We can add p-values from to the box plot of the distances:</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">get_pos_whisker</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co"># Calculate whisker position</span></span>
<span>  <span class="va">hinge</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span>, probs <span class="op">=</span> <span class="fl">0.75</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">upper_max</span> <span class="op">&lt;-</span> <span class="va">hinge</span> <span class="op">+</span><span class="fu"><a href="https://rdrr.io/r/stats/IQR.html" class="external-link">IQR</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">*</span><span class="fl">1.5</span> </span>
<span>  <span class="va">x</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://purrr.tidyverse.org/reference/keep.html" class="external-link">keep</a></span><span class="op">(</span><span class="op">~</span> <span class="va">.</span> <span class="op">&lt;</span> <span class="va">upper_max</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="co"># Calculate label position</span></span>
<span><span class="va">pos</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/by.html" class="external-link">by</a></span><span class="op">(</span><span class="va">mod_layer</span><span class="op">$</span><span class="va">distances</span>, <span class="va">mod_layer</span><span class="op">$</span><span class="va">group</span>, <span class="va">get_pos_whisker</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># You can run the next line for calculating automatically the</span></span>
<span><span class="co"># labels based in p-values.</span></span>
<span><span class="va">tukey_label</span> <span class="op">&lt;-</span>   <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">tukey_test</span><span class="op">$</span><span class="va">group</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span></span>
<span>      <span class="va">tukey_test</span><span class="op">$</span><span class="va">group</span><span class="op">[</span>,<span class="st">'p adj'</span><span class="op">]</span><span class="op">&lt;</span> <span class="fl">0.05</span></span>
<span>      <span class="op">)</span><span class="op">]</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate.html" class="external-link">separate</a></span><span class="op">(</span><span class="va">label</span>,sep <span class="op">=</span> <span class="st">'-'</span>, into <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'from'</span>, <span class="st">'to'</span><span class="op">)</span>,remove <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/exposition.html" class="external-link">%$%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>from <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">from</span>, <span class="va">to</span><span class="op">)</span>, to <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">to</span>, <span class="va">from</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">from</span><span class="op">)</span><span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">to</span>, collapse <span class="op">=</span> <span class="st">','</span><span class="op">)</span><span class="op">)</span><span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">'from'</span><span class="op">)</span><span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">label</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># However, for this data, there is no significance difference </span></span>
<span><span class="co"># between groups. We can still plot those labels as example:</span></span>
<span></span>
<span><span class="va">custom_label</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'A'</span>, <span class="st">'V'</span>, <span class="st">'AV'</span>, <span class="st">'T2'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span></span>
<span>  distances <span class="op">=</span> <span class="va">mod_layer</span><span class="op">$</span><span class="va">distances</span>,</span>
<span>  group <span class="op">=</span> <span class="va">mod_layer</span><span class="op">$</span><span class="va">group</span>,</span>
<span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">group</span>, y <span class="op">=</span> <span class="va">distances</span>, fill <span class="op">=</span> <span class="va">group</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_text</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>      distances <span class="op">=</span> <span class="va">pos</span>,group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pos</span><span class="op">)</span>,label <span class="op">=</span> <span class="va">custom_label</span></span>
<span>      <span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="va">label</span><span class="op">)</span>,</span>
<span>    hjust <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">0.5</span></span>
<span>    <span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">'Layer'</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">'Distance to Centroid'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">'None'</span><span class="op">)</span></span></code></pre></div>
<p><img src="phyloseq_analysis_files/figure-html/unnamed-chunk-35-1.png" width="680.314960629921"></p>
</div>
<div class="section level3">
<h3 id="annotating-nmds-with-information-from-permanova">Annotating NMDS with information from permanova<a class="anchor" aria-label="anchor" href="#annotating-nmds-with-information-from-permanova"></a>
</h3>
<ul>
<li><p>“Numbers on the top indicate the stress values.</p></li>
<li><p>Asterisks represent the significance level of PERMANOVA
(∗adjusted P &lt; 0.05 or ∗∗adjusted P &lt; 0.01)”</p></li>
</ul>
<p>First, we extract information from permanova</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span></span>
<span>  <span class="va">r2_label</span> <span class="op">&lt;-</span> <span class="va">permanova_layer</span><span class="op">$</span><span class="va">R2</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/format.html" class="external-link">format</a></span><span class="op">(</span>nsmall <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "0.42"</span></span>
<span><span class="op">(</span></span>
<span><span class="va">p_value_label</span> <span class="op">&lt;-</span> <span class="va">permanova_layer</span><span class="op">$</span><span class="va">`Pr(&gt;F)`</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="op">{</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html" class="external-link">case_when</a></span><span class="op">(</span></span>
<span>    <span class="va">.</span> <span class="op">&lt;</span> <span class="fl">0.01</span> <span class="op">~</span> <span class="st">"**"</span>,</span>
<span>    <span class="va">.</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">~</span> <span class="st">"*"</span>,</span>
<span>    <span class="va">.</span> <span class="op">&gt;=</span> <span class="fl">0.05</span> <span class="op">~</span> <span class="st">""</span>,</span>
<span>    <span class="op">)</span><span class="op">}</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "**"</span></span></code></pre></div>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">original_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/plot_ordination.html" class="external-link">plot_ordination</a></span><span class="op">(</span></span>
<span>  <span class="va">physeq_rel</span>, <span class="va">physeq_nmds</span>,</span>
<span>  type <span class="op">=</span><span class="st">"samples"</span>, color <span class="op">=</span> <span class="st">"layer"</span>,shape <span class="op">=</span> <span class="st">"layer"</span></span>
<span>  <span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/stat_ellipse.html" class="external-link">stat_ellipse</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>    legend.position<span class="op">=</span><span class="st">"bottom"</span>,</span>
<span>    legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="va">original_plot</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/annotate.html" class="external-link">annotate</a></span><span class="op">(</span></span>
<span>    geom<span class="op">=</span><span class="st">"text"</span>,</span>
<span>    x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">original_plot</span><span class="op">$</span><span class="va">data</span><span class="op">$</span><span class="va">NMDS1</span><span class="op">)</span> <span class="op">+</span> <span class="fl">0.4</span>,</span>
<span>    y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">original_plot</span><span class="op">$</span><span class="va">data</span><span class="op">$</span><span class="va">NMDS2</span><span class="op">)</span>,</span>
<span>    label<span class="op">=</span> <span class="va">r2_label</span>,</span>
<span>    size <span class="op">=</span> <span class="fl">5</span></span>
<span>    <span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/annotate.html" class="external-link">annotate</a></span><span class="op">(</span></span>
<span>    geom<span class="op">=</span><span class="st">"text"</span>,</span>
<span>    x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">original_plot</span><span class="op">$</span><span class="va">data</span><span class="op">$</span><span class="va">NMDS1</span><span class="op">)</span> <span class="op">+</span> <span class="fl">0.4</span>,</span>
<span>    y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="va">original_plot</span><span class="op">$</span><span class="va">data</span><span class="op">$</span><span class="va">NMDS2</span><span class="op">)</span>,</span>
<span>    label<span class="op">=</span> <span class="va">p_value_label</span>,</span>
<span>    size <span class="op">=</span> <span class="fl">10</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="phyloseq_analysis_files/figure-html/unnamed-chunk-37-1.png" width="680.314960629921"></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Francisco Campuzano.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
